### PR TITLE
Add reshape operations

### PR DESCRIPTION
**Pull request by arogowie-intel**
Add:
- Flatten
- Transpose
- Slice
- Concat
- Squeeze
- Unsqueeze
- Split

### CODE REVIEW

File: ngraph_onnx/onnx_importer/utils/reshape.py [hunk 1]
```diff
@@ -92,3 +113,41 @@ def flatten_innermost_empty_dims(node):  # type: (NgraphNode) -> NgraphNode
         return ng.reshape(node, input_order, output_shape)
     else:
         return node
+
+
+def get_bound(requested_bound, max_bound_val):  # type: (int, int) -> int
+    """Return valid bound value withing range [0,max_bound_val].
```
Comment (postrational) [inline ngraph_onnx/onnx_importer/utils/reshape.py]: What is a bound value? Please use a different term in docstring to explain what you mean.
Alternativel add another sentence to explain what bound means in this context.
What are we using this for?

File: tests/test_reshape.py [hunk 2]
```diff
@@ -32,13 +32,12 @@ def test_reshape():
     assert np.array_equal(ng_results, [expected_output])
 
 
-@pytest.mark.skip(reason='Needs refactoring to ngraph++')
 @pytest.mark.parametrize('axis,expected_output', [
-    (0, np.arange(120).reshape(120)),
+    (0, np.arange(120).reshape(1, 120)),
```
Comment (postrational) [inline tests/test_reshape.py]: Is this in line with ONNX backend tests for flatten?

File: ngraph_onnx/onnx_importer/ops_bridge.py [hunk 3]
```diff
@@ -438,101 +436,130 @@ def GlobalAveragePool(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphN
...
+    if axes is None:
+        raise ValueError('Unsqueeze node (%s): the "axes" attribute is mandatory.', onnx_node.name)
+
+    input_order = list(range(len(data.shape)))
+    out_shape = list(data.shape)
+    axes.sort()
+    for axe in axes:
+        # XXX: this condition forbids adding new dimensions greater than len(out_shape), i.e:
+        # if we have input tensor of shape (3,4,5) and we provide 'axes' attribute with value
+        # [10], then such input is considered invalid.
```
[the remark this answers is not in the document]
Comment (arogowie-intel) [reply ngraph_onnx/onnx_importer/ops_bridge.py]: There are already tests for `Unsqueeze` in ONNX. Also there is the case you provided.

File: tests/test_reshape.py (hunk 2 above)
Comment (arogowie-intel) [reply tests/test_reshape.py]: Yup, you can see it [here](https://github.com/onnx/onnx/blob/master/docs/Operators.md#attributes-17).

**Comment by arogowie-intel** [comment]
Updated.

File: ngraph_onnx/onnx_importer/utils/reshape.py [hunk 4]
```diff
@@ -92,3 +113,39 @@ def flatten_innermost_empty_dims(node):  # type: (NgraphNode) -> NgraphNode
...
+    :param node: The node we want to slice.
+    :param axes: The list of axes on which we perform slicing.
+    :param starts: The start index (inclusive) of slice for each sliced axis respectively.
+    :param ends: The end index (exclusive) of slice for each sliced axis respectively.
+    :return: The new node representing sliced portion of input node data.
+    """
+    lower_bounds = [0] * len(node.shape)
+    upper_bounds = list(node.shape)
+
+    for idx, axe in enumerate(axes):
```
Comment (postrational) [inline ngraph_onnx/onnx_importer/utils/reshape.py]: `axe` -> `axis`

Comment (arogowie-intel) [reply ngraph_onnx/onnx_importer/utils/reshape.py]: Fixed

**Pull request merged**